src/vos/README.md

```diff
@@ -105,79 +102,66 @@ For example, when an array object is "punched" from 0 to infinity in a given epo
+When performing lookup on a single value in an object, the object index is traversed to find the index node with the highest epoch number less than or equal to the requested epoch (near-epoch) that matches the key.
 If a value or negative entry is found, it is returned.
 Otherwise a "miss" is returned meaning that this key has never been updated in this VOS.
-This ensures that the most recent value in the epoch history of the KV is returned irrespective of the time-order in which they were integrated, and that all updates after the requested epoch are ignored.
+This ensures that the most recent value in the epoch history of is returned irrespective of the time-order in which they were integrated, and that all updates after the requested epoch are ignored.
 
 Similarly, when reading an array object, its index is traversed to create a gather descriptor that collects all object extent fragments in the requested extent with the highest epoch number less than or equal to the requested epoch.
 Entries in the gather descriptor either reference an extent containing data, a punched extent that the requestor can interpret as all zeroes, or a "miss", meaning that this VOS has received no updates in this extent.
 Again, this ensures that the most recent data in the epoch history of the array is returned for all offsets in the requested extent, irrespective of the time-order in which they were written, and that all updates after the requested epoch are ignored.
 
-A "miss" is distinct from a punched extent in an array object or a punched KV entry in a KV to support storage tiering at higher layers in the stack.
-This allows the VOS to be used as a cache since it indicates that the VOS has no history at this extent or key, and therefore data must be obtained from the cache's backing storage tier.
-
 <a id="711"></a>
 
 ### VOS Indexes
 
-Design of the object index and epoch index tables are similar to the container index table with the object ID and epoch number as keys respectively.
-The value of the object index table points to the location of the respective object structure.
-The value in case of the epoch index points to the object ID updated in that epoch.
+The value of the object index table points to a DKEY index.
+The values in the DKEY index, indexed by DKEY and epoch, point to an AKEY index.
+The values in the AKEY index, indexed by AKEY and epoch, point to either a Single Value index or an Array index.
+A single value index is referenced by epoch and will return the latest value inserted at or prior to the epoch.
+An array value is indexed by the extent and the epoch and will return portions of extents visible at the epoch.
 
-Objects can be either a key-value index data structure, byte-array index data structure, or a document store.
+Hints about the expectations of the object can be encoded in the object ID.
+For example, an object can be replicated, erasure coded, use checksums, or have integer or lexical DKEYs and/or AKEYs.
```

Comment:
checksums will be configured for containers, but I guess the object ID can encode the same info. All objects within a container will have the same checksum configuration.